athensresearch/athens
# chore: upgrade yarn deps alongside lein deps

close #1245

What this commit does:
- make sure `shadow-cljs` has same version for yarn (package.json) and lein (project.clj)
- make sure we are using `react17`/`react-dom17`, which is required by `reagent1.0.0`
- use latest `re-frame-10x` again
  - update `highlight.js` from 9 to 10
- run `yarn upgrade`

Also:
- Add `app` to builds so website is always built/compiled/released if renderer is
- Mirror `yarn` and `lein` commands, see https://github.com/athensresearch/athens/pull/1229#discussion_r640692486 cc @arkRedM 

General problems:
- It's possible for lein and yarn dependencies to get out of sync. e.g. could have 2.10.1 for yarn and 2.11.1 for lein - [source](https://github.com/thheller/shadow-cljs/issues/734#issuecomment-702050391)
- Some of our yarn dependencies have `^`, some don't. `^` means we can just minor and patch updates, not major. To confirm whether this was the case here, but it's possible updating minor or patch version still led to breaking changes - [source](https://stackoverflow.com/a/22345808)
- It's possible for lein dependencies to require incompatible versions of yarn dependencies, as was the case with highlighter
- It's possible for compilation to work for `lein compile` but not `lein prod`, because `lein prod` minifies code. This is why desktop build was working, but didn't realize until prod demo was updated that the website wasn't working.

Going forward:
- QA for both desktop and website before pushing a new tag/release
- Pay closer to attention to compile errors, we actually still have some. Some warnings should really be errors, letting builds pass incorrectly.
- Be careful when updating dependencies. #1216 did a lot of updates and it was naive to think it would be perfectly fine! npm ecosystem known to be fragile and shadow interacts with it in an interesting way
